egs/wsj/s5/steps/limit_num_gpus.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+# This script functions as a wrapper of a bash command that uses GPUs.
+#
+# It sets the CUDA_VISIBLE_DEVICES variable so that it limits the number of GPUs
+# used for programs. It is neccesary for running a job on the grid if the job
+# would automatically grabs all resources available on the system, e.g. a
+# TensorFlow program.
+
+num_gpus=1
+
+[ -f ./path.sh ] && . ./path.sh; # source the path.
+. parse_options.sh || exit 1;
+
+CUDA_VISIBLE_DEVICES=
+num_total_gpus=`nvidia-smi -L | wc -l | awk '{print $1}'`
+num_gpus_assigned=0
+
+for i in `seq 0 $[$num_total_gpus-1]`; do
+# going over all GPUs and check if it is idle, and add to the list if yes
+  idle=`nvidia-smi -i $i | grep "No running processes found" | wc -l | awk '{print $1}'`
```

Comment:
I'm not sure that this would be sufficiently fast if nvidia-persistenced were not running, but anyway, we'll address this later if it's an issue.  The awk here is not needed since wc is running from the stdin.
Actually you can do just do:
```
  if nvidia-smi -i $i | grep "No running processes found" >/dev/null; then
    # do something
  fi
```
instead of setting the idle variable.  grep returns 1 if it mached no lines, 0 otherwise.